apps/trading/client-pages/market/market.tsx

```diff
@@ -51,63 +59,17 @@ export const Market = () => {
     [marketId, navigate]
   );
 
-  const variables = useMemo(
-    () => ({
-      marketId: marketId || '',
-    }),
-    [marketId]
-  );
-
-  const updateMarketId = useCallback(
-    ({ data }: { data: { id?: string } | null }) => {
-      if (data?.id && data.id !== lastMarketId) {
-        update({ marketId: data.id });
-      }
-      return true;
-    },
-    [update, lastMarketId]
-  );
-  const { data, error, loading } = useDataProvider<
-    SingleMarketFieldsFragment,
-    never
-  >({
-    dataProvider: marketProvider,
-    variables,
-    update: updateMarketId,
-    skip: !marketId,
-  });
-
-  const marketName = data?.tradableInstrument.instrument.name;
-  const updateProvider = useCallback(
-    ({ data: marketData }: { data: MarketData | null }) => {
-      const marketPrice = calculatePrice(
-        marketData?.markPrice,
-        data?.decimalPlaces
-      );
-      if (marketName) {
-        const newPageTitle = titlefy([marketName, marketPrice]);
-        if (pageTitle !== newPageTitle) {
-          updateTitle(newPageTitle);
-        }
-      }
-      return true;
-    },
-    [updateTitle, pageTitle, marketName, data?.decimalPlaces]
-  );
-
-  useDataProvider<MarketData, MarketDataUpdateFieldsFragment>({
-    dataProvider: marketDataProvider,
-    update: updateProvider,
-    variables,
-    skip: !marketId || !data,
-  });
-
   const tradeView = useMemo(() => {
+    if (!data?.market) {
+      return null;
+    }
+
     if (w > 960) {
-      return <TradeGrid market={data} onSelect={onSelect} />;
+      return <TradeGrid market={data.market} onSelect={onSelect} />;
```

Comment:
When we passing down to TradePanels constantly differentiating market object, we can cause a massive rerender. That's why previously we passed only constant _Market_ object with no changeable _MarketData_ part.

Reply:
I totally agree with @macqbat.
We on purpose separated market from market data.
Market don't change and it was passed down as a prop.
All components that were using market data were subscribing to updates through data provider and was reusing same data from provider. 
We avoid rerender of whole TradeGrid on marketData change. 
Additionally all components (such as MarketMarkPrice, MarketTradingMode or MarketVolume) that were using market data can be reused in any other place providing only marketId in props.  
It was all replaced by MarketHookData with subscribeToMore.

Reply:
The market object here doesn't make it down to any of the windows, only really to TradeGridHeader. Each window only takes a marketId (I think they all do but need to double check) and should manage their own query, as they do now

Reply:
_Why do components get re-rendered?

[...]

Re-rendering of parent component: Whenever the components render function is called, all its subsequent child components will re-render, regardless of whether their props have changed or not._

If market data will change if will cause all sub components re render. We are connecting only very end components to 
live market data to avoid re rendering where it's not needed.
Even more. Components which are connected to market data subscription  (like MarketMarkPrice or  MarketVolume) are will  re render only when data significant to what they render will change and this update is throttled.